@here/harp-mapview/lib/geometry/SolidLineMesh.ts

```diff
@@ -0,0 +1,451 @@
+        : Infinity;
+    sphere.center.copy(line.end);
+    const endCapT = sphere.containsPoint(ray.origin)
+        ? 0
+        : ray.intersectSphere(sphere, tmpV4)
+        ? tmpV4.sub(ray.origin).length()
+        : Infinity;
+    return Math.min(startCapT, endCapT);
+}
+
+/**
+ * Intersects line
+ * @param ray Intersection ray in object local space.
+ * @param line The line to intersect.
+ * @param vExtrusion Line extrusion vector.
+ * @param hWidth The line's extrusion half width.
+ * @param hWidthSq The line's extrusion half width squared.
+ * @param plane The extrusion plane.
+ * @param interPlane The intersection of the ray with the extrusion plane.
+ * @param interLine The ray intersetion with the extruded line.
+ * @returns true if ray intersects the extruded line, false otherwise.
+ */
+function intersectLine(
+    ray: THREE.Ray,
+    line: THREE.Line3,
+    vExtrusion: THREE.Vector3,
+    hWidth: number,
+    hWidthSq: number,
+    plane: THREE.Plane,
+    interPlane: THREE.Vector3,
+    interLine: THREE.Vector3
+): boolean {
+    if (interPlane.equals(tmpRay.origin) && tmpRay.direction.dot(plane.normal) === 0) {
+        // Corner case: ray is coplanar to extruded line, find distance to extruded line sides
+        // and end caps.
+        const extrLineT = intersectExtrudedLine(ray, line, vExtrusion, plane.normal, hWidth);
+        const endCapT = intersectClosestEndCap(ray, line, hWidth);
+
+        const minT = Math.min(extrLineT, endCapT);
+        if (minT === Infinity) {
+            return false;
+        }
+        ray.at(minT, interLine);
+        return true;
+    }
+
+    // The plain intersection is also a line intersection only if it's closer to the line
+    // than the extrusion half width.
+    const distSq = interPlane.distanceToSquared(line.closestPointToPoint(interPlane, true, tmpV4));
+
+    if (distSq > hWidthSq) {
+        return false;
+    }
+    interLine.copy(interPlane);
+    return true;
+}
+
+/**
+ * Finds the intersections of a ray with a partition of a solid line mesh representing a feature.
+ * @param mesh The mesh whose intersections will be found.
+ * @param raycaster Contains the intersection ray.
+ * @param localRay Same ray as raycaster.ray but in object local space.
+ * @param intersects Array that will contain all intersections found between ray and feature.
+ * @param halfWidth The line's extrusion half width.
+ * @param lHalfWidth The line's extrusion half width in mesh local space.
+ * @param lHalfWidthSq The line's extrusion half width squared in mesh local space.
+ * @param beginIdx The index where the feature starts in the mesh geometry's indices attribute.
+ * @param endIdx The index where the feature end in the mesh geometry's indices attribute.
+ * @param bSphere The feature bounding sphere.
+ * @returns feature
+ */
+function intersectFeature(
+    mesh: THREE.Mesh,
+    raycaster: THREE.Raycaster,
+    localRay: THREE.Ray,
+    intersects: THREE.Intersection[],
```

Comment:
Better to move output params to the end... and note in documentation that this is output.

Reply:
done.